src/lib/operators/join_hash.cpp

```diff
@@ -597,8 +604,53 @@ class JoinHash::JoinHashImpl : public AbstractJoinOperatorImpl {
         _mode(mode),
         _column_ids(column_ids),
         _predicate_condition(predicate_condition),
-        _inputs_swapped(inputs_swapped),
-        _radix_bits(radix_bits) {}
+        _inputs_swapped(inputs_swapped) {
+    /*
+            Hash joins perform best for join relations with a small left join partner. In case the 
+            optimizer selects the hash join due to such a situation, but neglects that the
+            input will be switched (e.g., due to the join type), the user will be warned.
+          */
+    if (inputs_swapped) {
+      PerformanceWarning("Inputs swapped for hash join.");
+    }
+
+    /*
+            Setting number of bits for radix clustering:
+            The number of bits is used to create probe partitions with a size that can
+            be expected to fit into the L2 cache.
+            This should incorporate hardware knowledge, once available in Hyrise.
+            As of now, we assume a L2 cache size of 256 KB.
+            We estimate the size the following way:
+              - we assume each key appears once (that is an overestimation space-wise, but we
+              aim rather for a hash map that is slightly smaller than L2 than slightly larger)
+              - each entry in the hash map is a data structure holding the actual value
+              and the RowID
```

Comment:
This is false with the current implementation where multiple row ids may be stored with the same key. Since this is a simple estimation, we might not have to change it, but we should at least reflect it in the comment.

Reply:
Ah, even your calculation below tells a different story.

Reply:
Yeah, I will clear that up. The calculation ought to be some kind of a worst case calculation.